### PR TITLE
Allow to hook context factories

### PR DESCRIPTION
**Pull request by thc202**
Change classes ExtensionHook and ExtensionHookView to allow to add
ContextDataFactory and ContextPanelFactory, respectively, so the
factories can be automatically added/removed by core code.
Change ExtensionLoader to add/remove the context factories when the
extensions are initialised/unloaded.

### CODE REVIEW

**Comment by thc202** [comment]
Convenience changes for #2379 and #2380.

File: src/org/parosproxy/paros/extension/ExtensionHookView.java [hunk 1]
```diff
@@ -80,4 +94,32 @@ public void addOptionPanel(AbstractParamPanel panel) {
         return optionPanelList;
     }
     
+    /**
+     * Adds the given {@link ContextPanelFactory} to the view hook, to be latter added to the
```
Comment (kingthorin) [inline src/org/parosproxy/paros/extension/ExtensionHookView.java]: I think you want later not latter

File: src/org/parosproxy/paros/extension/ExtensionHook.java [hunk 2]
```diff
@@ -189,4 +202,32 @@ public ViewDelegate getView() {
     public List<OverrideMessageProxyListener> getOverrideMessageProxyListenerList() {
         return overrideMessageProxyListenersList;
     }
+
+    /**
+     * Adds the given {@link ContextDataFactory} to the extension hook, to be latter added to the {@link Model}.
```
Comment (kingthorin) [inline src/org/parosproxy/paros/extension/ExtensionHook.java]: I think you want later not latter

**Comment by kingthorin** [comment]
Two minor java doc comments, otherwise looks good.

**Comment by thc202** [comment]
Fixed, thank you.

**Pull request merged**